src/libvirt/Connect.cs

```diff
@@ -19,13 +19,13 @@ public Connect(string uri)
 
         public string Uri { get; }
 
-        public bool IsOpen { get; set; }
+        public bool IsOpen { get; private set; }
 
         public void Open(bool readOnly = false)
         {
             if (IsDisposed)
             {
-                throw new ObjectDisposedException("Cannot open a disposed Connect");
+                throw new ObjectDisposedException(null, "Cannot open a disposed Connect.");

```

Comment:
Why this overload?

Reply:
```
//
// Summary:
//     Initializes a new instance of the System.ObjectDisposedException class with the
//     specified object name and message.
//
// Parameters:
//   objectName:
//     The name of the disposed object.
//
//   message:
//     The error message that explains the reason for the exception.
public ObjectDisposedException(string? objectName, string? message);
```

The `objectName` is often an argument name etc, in which there is no object name here.

Reply:
You are right. Consider passing this.GetType().Name or anything else indicating at least the object "type". Null is ok anyway.

Reply:
I've used "Connect" as first parameter, please take a look on main branch